examples/plot_metadata_routing.py

```diff
@@ -0,0 +1,624 @@
+            ignore_extras=False, kwargs=fit_params
+        )
+        self.estimator_ = clone(self.estimator).fit(X, y, **fit_params_)
+        self.classes_ = self.estimator_.classes_
+        return self
+
+    def predict(self, X, **predict_params):
+        check_is_fitted(self)
+        # same as in ``fit``, we validate the given metadata
+        metadata_request_factory(self).predict.validate_metadata(
+            ignore_extras=False, kwargs=predict_params
+        )
+        # and then prepare the input to the underlying ``predict`` method.
+        predict_params_ = metadata_request_factory(
+            self.estimator_
+        ).predict.get_method_input(ignore_extras=False, kwargs=predict_params)
+        return self.estimator_.predict(X, **predict_params_)
+
+    def get_metadata_request(self):
+        router = (
+            MetadataRouter()
+            .add(super(), mapping="one-to-one", overwrite=False, mask=False)
+            .add(self.estimator, mapping="one-to-one", overwrite="smart", mask=True)
+        )
+        return router.get_metadata_request()
+
+
+# %%
+# The two key parts where the above estimator differs from our previous
+# meta-estimator is validation in ``fit``, and generating routing data in
+# ``get_metadata_request``. In ``fit``, we pass ``self_metadata=super()`` to
+# ``validate_metadata``. This is important since consumers don't validate how
+# metadata is passed to them, it's only done by routers. In this case, this
+# means validation should be done for the metadata consumed by the
+# sub-estimator, but not for the metadata consumed by the meta-estimator
+# itself.
+#
+# In ``get_metadata_request``, we add what's consumed by this meta-estimator
+# without masking them, before adding what's requested by the sub-estimator.
+# Passing ``super()`` here means only what's explicitly mentioned in the
+# methods' signature is considered as metadata consumed by this estimator; in
+# this case fit's ``foo``. Let's see what the routing metadata looks like with
+# different settings:
+
+# %%
+# no metadata requested
+est = RouterConsumerClassifier(estimator=ExampleClassifier())
+est.get_metadata_request()["fit"]
+
+
+# %%
+# ``foo`` requested by child estimator
+est = RouterConsumerClassifier(estimator=ExampleClassifier().fit_requests(foo=True))
+est.get_metadata_request()["fit"]
+# %%
+# ``foo`` requested by meta-estimator
+est = RouterConsumerClassifier(estimator=ExampleClassifier()).fit_requests(foo=True)
+est.get_metadata_request()["fit"]
+
+# %%
+# As you can see, the last two are identical, which is fine since that's what a
+# meta-estimator having ``RouterConsumerClassifier`` as a sub-estimator needs.
+# The situation is different if we use named aliases:
+#
+# Aliased on both
+est = RouterConsumerClassifier(
+    foo_is_none=False,
+    estimator=ExampleClassifier(foo_is_none=False).fit_requests(
+        foo="first_aliased_foo"
```

Comment:
there is maybe better than first and second and being explicit regarding the consumer